packages/admin-panel/src/editor/EditModal.js

```diff
@@ -43,12 +43,26 @@ export const EditModalComponent = ({
   usedBy,
   usedByIsLoading,
   usedByErrorMessage,
+  endpoint,
 }) => {
   const fieldsBySource = keyBy(fields, 'source');
 
+  const getTitle = () => {
+    if (endpoint !== null && Array.isArray(title)) {
+      const { titleText: selectedTitle } = title
+        .filter(config => endpoint.includes(config.titleEndpoint))
+        .shift();
+      return selectedTitle;
```

Comment:
This is a nice solution, but it might be worth a call to snoop around what else could be done to solve this - let me know if you're keen for that tomorrow :-)

Reply:
Yep, thanks. will call you when I've done some other priority items. @edmofro